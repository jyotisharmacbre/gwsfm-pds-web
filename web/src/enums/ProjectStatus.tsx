enum ProjectStatus {
  InitialCustomerInquiry = 1,
  JA = 2,
  InReview = 3,
  BidLost = 6,
  OnHold = 7,
  OrderReceived = 5,
  JAApproved = 4,
  Live = 9,
  Completed = 10
}

export default ProjectStatus;
