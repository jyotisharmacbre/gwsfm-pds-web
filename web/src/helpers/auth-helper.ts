// @ts-ignore
import authentication from '@kdpw/msal-b2c-react';
import jwtDecode from 'jwt-decode';
import { DecodedToken } from '../models/decodedToken';

export const getDisplayName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.name;
};

export const getFirstName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.given_name;
};

export const getDisplayEmail = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.email ?
    decoded.email :
    decoded.emails && decoded.emails.length > 0 ?
      decoded.emails[0] :
      '';
};

export const logOut = () => {
  authentication.signOut();
}

function decodeToken() {
  const token = authentication.getAccessToken();
  const decoded: DecodedToken = jwtDecode(token);
  return decoded;
}

