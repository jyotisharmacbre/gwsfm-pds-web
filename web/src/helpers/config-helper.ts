const appConfig = () => {
    const config: any = (window as any).Config;
    if(config && config.REACT_APP_AUTH_INSTANCE==="#{AUTH_INSTANCE}#"){
      return {
        ...process.env
      }
    }
    return {
      ...config
    }
  };
  
  export default appConfig;