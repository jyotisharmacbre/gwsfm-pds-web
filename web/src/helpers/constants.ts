export const selectionButtons = [
  {
    title: 'BUTTON_YES',
    value: true
  },
  {
    title: 'BUTTON_NO',
    value: false
  }
];

export enum Notify {
  none,
  success,
  error
}
