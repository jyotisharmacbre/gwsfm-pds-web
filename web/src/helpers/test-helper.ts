import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
