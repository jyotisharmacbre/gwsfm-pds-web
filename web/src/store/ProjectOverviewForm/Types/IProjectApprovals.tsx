export interface IProjectApprovals {
      projectApprovalId: string;
      projectId: string;
      projectApprovalRange: string;
      approverType: string;
      approvalStatus: string;
      userId: string;
  }