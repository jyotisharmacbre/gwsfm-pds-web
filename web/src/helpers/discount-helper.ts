import DiscountType from '../enums/DiscountType';
import { restrictMinusAndAllowDecimalForMaxRangeHundred, restrictMinusAndAllowDecimal } from '../helpers/utility-helper';

const normalizeClientDiscount = (discountType: DiscountType, discountValue) => {
    let updatedDiscount;
    if (discountType === DiscountType.Percent) {
        updatedDiscount = restrictMinusAndAllowDecimalForMaxRangeHundred(discountValue);
    }
    else updatedDiscount = restrictMinusAndAllowDecimal(discountValue);
    return updatedDiscount;
}

export default normalizeClientDiscount;