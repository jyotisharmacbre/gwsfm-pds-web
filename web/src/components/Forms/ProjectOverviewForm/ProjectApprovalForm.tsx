import React from 'react';
import { FormattedMessage } from 'react-intl';
import NewTypeAhead from '../../TypeAhead/NewTypeAhead';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheckCircle, faClock, faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
import { ProjectSignOffStatus } from '../../../store/ProjectOverviewForm/Types/ProjectApprovalEnums';
import ProjectStatus from '../../../enums/ProjectStatus';

interface IProps {
	fields: any;
	formatUserData: any;
	getListOfUsers: (value: any) => Promise<any>;
	status: number
}

const ProjectApprovalForm: React.FC<IProps> = (props) => {
	const { fields, formatUserData, getListOfUsers } = props;
	const getClassAndIcon = (approverType) => {
		let className: string = '', iconType: any;
		let labelID: string = '';
		switch (approverType) {
			case ProjectSignOffStatus.Approved:
				className = "green";
				iconType = faCheckCircle;
				labelID = "LABEL_APPROVED"
				break;
			default:
				className = "orange";
				iconType = faClock;
				labelID = "LABEL_PENDING";
				break;
		}
		return {
			className: className,
			iconType: iconType,
			labelKey: labelID
		};
	}
	const shouldDisplay = (currVal) => {
		return currVal.userId && currVal.projectApprovalId && props.status == ProjectStatus.InReview
	};
	return (
		<div>
			<div className="row">
				<div className="col-md-12 d-flex">
					<label />
					<h6 className="mb-0 d-none">
						<FormattedMessage id="LABEL_SIGN_OFF_STATUS" />{' '}
					</h6>
				</div>
			</div>
			{fields.map((member, index) => (
				<div className="row align-items-stretch" key={index} data-test="project-approval-form">
					<div className="col-11">
						<div className="form-group">
							<NewTypeAhead
								name={`${member}.userId`}
								onSearch={getListOfUsers}
								formatData={formatUserData}
								DynamicsType={fields.get(index).approverTypeDescription}
								labelName={
									fields.get(index).showRangeLabel ? (
										fields.get(index).projectApprovalRangeDescription
									) : null
								}
								submitParam="email"
							/>
							<span className="right_fix_txt">{fields.get(index).approverTypeDescription}</span>
						</div>
					</div>
					<div className="col-1 pl-0">
						{shouldDisplay(fields.get(index)) && <div className="approve_state ">
							<div className="form-group d-flex">
							{<span data-test="icons-approval-status" className="icon"><FontAwesomeIcon className={getClassAndIcon(fields.get(index).approvalStatus).className} icon={getClassAndIcon(fields.get(index).approvalStatus).iconType} /></span>}
							{<label className='approv_label'><FormattedMessage id={getClassAndIcon(fields.get(index).approvalStatus).labelKey} /> </label>}
						</div></div>
						}	</div>
				</div>
			))}
		</div>
	);
};

export default ProjectApprovalForm;
