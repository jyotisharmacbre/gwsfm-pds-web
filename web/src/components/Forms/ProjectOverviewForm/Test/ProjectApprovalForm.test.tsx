import { shallow } from 'enzyme';
import React from 'react';
import { findByTestAtrr } from '../../../../helpers/test-helper';
import { initialState } from '../../../../store/ProjectOverviewForm/Test/Reducertestdata';
import ProjectApprovalForm from '../ProjectApprovalForm';
import ProjectStatus from '../../../../enums/ProjectStatus';
import { IProjectApprovals } from '../../../../store/ProjectOverviewForm/Types/IProjectApprovals';

describe('Project Approval Form tests', () => {
    let testData = initialState.form.projectApprovals;
    let wrapper: any;
    let props: any;
    beforeEach(() => {
        props = {
            fields: {
                map: (callback) => testData.map((field, index) => callback(field, index)),
                getAll: jest.fn(x => testData),
                get: index => testData[index],
                length: testData.length,
            },
            formatUserData: jest.fn(),
            getListOfUsers: jest.fn(),
            status: ProjectStatus.InReview
        };


        wrapper = shallow(<ProjectApprovalForm
            {...props}
        />);

    });

    it('Defines the component', () => {
        expect(wrapper).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('renders the ProjectApprovalForm component with no errors', () => {
        expect(findByTestAtrr(wrapper, 'project-approval-form').length).toEqual(11);
    });
    it('should not render the icons against approvers if status is not in-review', () => {
        props.status = ProjectStatus.JA;
        wrapper = shallow(<ProjectApprovalForm {...props}
        />);
        expect(findByTestAtrr(wrapper, 'icons-approval-status').length).toEqual(0);
    });
    it('should render the icons against approvers if status is JAApproved', () => {
     testData[0] = {projectApprovalId : '1',
       projectId : '1',
       projectApprovalRange : 1,
       projectApprovalRangeDescription : '1',
       approverType: 1,
       approverTypeDescription: 'string',
       approvalStatus: 1,
       approvalStatusDescription: 'string',
       userId: 'string',
       showRangeLabel: true,

    }
        
        props.status = ProjectStatus.JAApproved;
        wrapper = shallow(<ProjectApprovalForm {...props}
        />);
        expect(findByTestAtrr(wrapper, 'icons-approval-status').length).toBeGreaterThan(0);
    });


});
