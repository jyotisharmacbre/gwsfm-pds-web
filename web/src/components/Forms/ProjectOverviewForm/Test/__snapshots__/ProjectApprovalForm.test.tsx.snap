// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Approval Form tests should match the snapshot 1`] = `ShallowWrapper {}`;
